pkg/controller/sparkapplication/submission_cache.go

```diff
@@ -0,0 +1,66 @@
+package sparkapplication
+
+/*
+A simple cache implementation that stores keys with a time-to-live (TTL) value.
+*/
+
```

Comment:
I create own custom and simple cache to avoid third libs and change the go mod

Reply:
I prefer that approach for simple things like that to be honest.

Reply:
We already use `github.com/jellydator/ttlcache/v3` in every other DP component
Go being a language that never break old code make relying on 3rd party pretty safe

Reply:
the main point is to avoid to update the `go.mod` file by addind third part, 
since this repo is a fork , once we want to rebase to the `main` of the original repo,
that will be more smooth

imo.

Reply:
We are already too far for that but i understand the logic behind this

Reply:
I agree with Seb, we use the `ttlcache` in our services and it will be natural to use it here. We are pretty far from the origin repo and this particular thing won't help us to reduce the amount of work. Even more, the spark-operator is moving forward very fast right now under the kubeflow, so merging won't be easy anyway.

Reply:
so will use `github.com/jellydator/ttlcache/v3` then